src/Files.App/Helpers/NavigationHelpers.cs

```diff
@@ -330,6 +331,27 @@ private static async Task<FilesystemResult> OpenDirectory(string path, IShellPag
 
 				opened = (FilesystemResult)true;
 			}
+			else if (isNetwork)
+			{
+				var auth = await NetworkDrivesAPI.AuthenticateNetworkShare(path);
+				if (auth)
+				{
+					if (forceOpenInNewTab || userSettingsService.FoldersSettingsService.OpenFoldersInNewTab)
```

Comment:
Do you think these can be merged to a single if statement? The else can be changed to an `else if`.